tests/test_cli.py

```diff
@@ -235,3 +249,46 @@ def predict(self, context, model_input):  # pylint: disable=unused-variable
     assert scoring_response.status_code == 200
     served_model_preds = np.array(json.loads(scoring_response.content))
     np.testing.assert_array_equal(served_model_preds, model.predict(data, None))
+
+
+def test_mlflow_tracking_disabled_in_artifacts_only_mode():
+
+    port = get_safe_port()
+    cmd = ["mlflow", "server", "--port", str(port), "--artifacts-only"]
+    process = subprocess.Popen(cmd)
+    with pytest.raises(
+        HTTPError,
+        match="Endpoint disabled due to the mlflow server running in `--artifacts-only` mode.",
+    ):
+        _await_server_up_or_die(port, timeout=10)
+        resp = requests.get(f"http://localhost:{port}/api/2.0/mlflow/experiments/list")
+        augmented_raise_for_status(resp)
```

Comment:
```suggestion
    _await_server_up_or_die(port, timeout=10)
    with pytest.raises(
        HTTPError,
        match="Endpoint disabled due to the mlflow server running in `--artifacts-only` mode.",
    ):
        resp = requests.get(f"http://localhost:{port}/api/2.0/mlflow/experiments/list")
        augmented_raise_for_status(resp)
```

Can we refactor the code here to clarify which line throws an `HTTPError`?

Reply:
```suggestion
    _await_server_up_or_die(port, timeout=10)
    resp = requests.get(f"http://localhost:{port}/api/2.0/mlflow/experiments/list")
    assert resp.status_code == 503
    assert "Endpoint disabled due to the mlflow server running in" in resp.text
```

On second thought, can we just check `resp.text` contains `Endpoint disabled due to the mlflow server running in`?

Reply:
```suggestion
    _await_server_up_or_die(port, timeout=10)
    resp = requests.get(f"http://localhost:{port}/api/2.0/mlflow/experiments/list")
    assert resp.status_code == 503
    assert "Endpoint disabled due to the mlflow server running in `--artifacts-only" in resp.text
```

On second thought, can we just check `resp.text` contains `Endpoint disabled due to the mlflow server running in`?

Reply:
changed that test and the test `test_mlflow_artifact_service_unavailable_without_config()`